### PR TITLE
Add joblisting notifications

### PR DESCRIPTION
**Pull request by LudvigHz**
Example image:
image
(The images are fixed by https://github.com/webkom/lego/pull/2962)

If anyone has ideas for better layouts/UI, I'm all ears!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -98,17 +98,41 @@ function getActiveFrom(pool) {
...
+function getJobYears(fromYear, toYear) {
+  return fromYear === toYear ? `${fromYear}. Klasse` : `${fromYear}. - ${toYear}. Klasse`;
+}
+
+async function retrieveJoblistings() {
+  const today = new Date();
+  const yesterday = new Date().setDate(today.getDate() - 1);
+  const date = dateFns.format(yesterday, 'YYYY-MM-DD');
+  const qs = querystring.stringify({ created_after: date });
+  let res = await callAPI(`${API_URL}/joblistings/?${qs}`, accessToken);
```
Comment (AdrianAndersen) [inline index.js]:
```suggestion
  const res = await callAPI(`${API_URL}/joblistings/?${qs}`, accessToken);
```

File: index.js [hunk 2]
```diff
@@ -124,36 +148,121 @@ function buildAttachments(events) {
...
-      thumb_url: event.cover
+      thumb_url: event.cover,
     };
   });
 }
 
+function buildJoblistingBlocks(joblistings) {
+  const blocks = joblistings.flatMap((joblisting, i) => {
+    const pretext = i === 0 ? 'Nye jobbanonser:' : '';
+
```
Comment (AdrianAndersen) [inline index.js]: The variable `pretext` is not in use in this scope.

File: index.js [hunk 3]
```diff
@@ -124,36 +148,121 @@ function buildAttachments(events) {
...
+  const webhook = new IncomingWebhook(process.env.WEBHOOK_URL_JOBLISTINGS, {
+    username: 'Abakus',
+    icon_url: 'https://abakus.no/icon-512x512.png',
+  });
+
+  webhook.send = promisify(webhook.send);
+  const blocks = buildJoblistingBlocks(
+    joblistings.slice(joblistings.length - 4, joblistings.length - 1)
+  );
+  console.log(JSON.stringify({ text: 'Nye jobbanonser i dag', blocks }));
```
Comment (AdrianAndersen) [inline index.js]: Is this necessary for logs, or is it just forgotten?

File: index.js [hunk 4]
```diff
@@ -124,36 +148,121 @@ function buildAttachments(events) {
...
     if (events.length > 0) {
-      events.forEach(event => console.log(` - ${event.title} `));
+      events.forEach((event) => console.log(` - ${event.title} `));
       await notifySlack(events);
     }
+
+    const joblistings = await retrieveJoblistings();
+    if (joblistings.length > 0) {
+      joblistings.forEach((joblisting) =>
+        console.log(` - ${joblisting.company.name}: ${joblisting.title}`)
```
Comment (AdrianAndersen) [inline index.js:260]: Is this also necessary?

File: index.js (hunk 3 above)
Comment (LudvigHz) [reply index.js]: Whoops, yeah this doesn't need to be here

File: index.js (hunk 4 above)
Comment (LudvigHz) [reply index.js:260]: This is for logging yeah

**Pull request merged**